mrf/js/56-data-fetching/mian.js

```diff
@@ -0,0 +1,99 @@
+// Define the API URL
+const apiUrl = "https://jsonplaceholder.typicode.com/";
+
+async function getAuthorPost(userID) {
+  const response = await fetch(apiUrl + `users/${userID}`);
+  const final = await response.json();
+
+  return final["name"];
+}
+
+async function getPost(id) {
+  const response = await fetch(apiUrl + `posts/${id}`);
+  const final = await response.json();
+
+  const author = await getAuthorPost(final["userId"]);
+  final["name"] = author;
+
+  return final;
+}
+
+allDivs = {};
+
+function addCard(titleIn, textIn, authorIn, ID) {
+  const master = document.createElement("div");
+  master.setAttribute("class", "card");
+
+  const title = document.createElement("p");
+  title.textContent = titleIn;
+
+  const text = document.createElement("p");
+  text.textContent = textIn;
+
+  const author = document.createElement("p");
+  author.textContent = authorIn;
+
+  const deleteButton = document.createElement("button");
+  deleteButton.setAttribute("class", "delete");
+  deleteButton.setAttribute("onClick", `deleteCard(${ID})`);
+  deleteButton.textContent = "Delete";
+
+  const updateButton = document.createElement("button");
+  updateButton.setAttribute("class", "update");
+  updateButton.setAttribute("onClick", `updateCard(${ID})`);
```

Comment:
Instead of using setAttribute, you can also use the addEventListener method.

Reply:
got it